File: src/main/java/me/andre111/d20server/scripting/variable/parser/VariableParserParentActorToken.java
package me.andre111.d20server.scripting.variable.parser;

import me.andre111.d20server.scripting.ActorFinderToken;
import me.andre111.d20server.scripting.ParserContext;

public class VariableParserParentActorToken extends VariableParserParent {
	@Override
	protected void updateContext(ParserContext context, String name) {
		//TODO: check for TokenFinder present!
		context.setActorFinder(new ActorFinderToken(context.getTokenFinder()));
	}
}

File: core/common/scripting/variable/parser/variable-parser-parent-actor-token.js
import { VariableParserParent } from './variable-parser-parent.js';
import { EntityFinderActorToken } from '../../finder/entity-finder-actor-token.js';

export class VariableParserParentActorToken extends VariableParserParent {
    constructor() {
        super();
    }

    updateContext(context, name) {
        if(!context.getEntityFinder('token')) throw new Error(`Internal Error: No EntityFinder for token present`);

        context.setEntityFinder('actor', new EntityFinderActorToken(context.getEntityFinder('token')));
    }
}
